test/resources/3scale_api.go

```diff
@@ -203,3 +208,119 @@ func (r *ThreeScaleAPIClientImpl) CreateProduct(name string) (string, error) {
 	id := strings.Split(href, "/")
 	return id[len(id)-1], nil
 }
+
+func (r *ThreeScaleAPIClientImpl) SetUserAsAdmin(username string, email string, userID string) error {
+
+	url := fmt.Sprintf("%v/p/admin/account/users/"+userID, r.host)
+	formUrl := fmt.Sprintf("%v/p/admin/account/users/%v/edit", r.host, userID)
+
+	// First try to get the CRSF token by requesting the form
+	csrf, err := requestCRSFToken(r.client, formUrl)
+	if err != nil {
+		return err
+	}
+
+	formValues := url2.Values{
+		"user[username]":              []string{username},
+		"user[email]":                 []string{email},
+		"user[role]":                  []string{"admin"},
+		"user[password]":              []string{},
+		"user[password_confirmation]": []string{},
+		"commit":                      []string{"Update+User"},
+		"authenticity_token":          []string{csrf},
+		"_method":                     []string{"patch"},
+	}
+
+	req, err := http.NewRequest(http.MethodPost, url, strings.NewReader(formValues.Encode()))
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %v", r.token))
+
+	resp, err := r.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return errors.New(fmt.Sprintf("expected 200 but got %v", resp.StatusCode))
+	}
+
+	return nil
+}
+
+func (r *ThreeScaleAPIClientImpl) GetUserId(username string) (string, error) {
+	url := fmt.Sprintf("%v/p/admin/account/users", r.host)
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %v", r.token))
+	resp, err := r.client.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "", errors.New(fmt.Sprintf("expected 200 but got %v", resp.StatusCode))
+	}
+
+	doc, err := ParseHtmlResponse(resp)
+	if err != nil {
+		return "", err
+	}
+
+	links := doc.Find("a[title='Edit']")
+
+	link := ""
+	links.Each(func(index int, s *goquery.Selection) {
+		if s.Contents().Text() == username {
+			link, _ = s.Attr("href")
+			return
+		}
+	})
+
+	if link == "" {
+		return "", fmt.Errorf("Failed to retrieve link to edit user")
+	}
+
+	userId := ""
+	s := strings.Split(link, "/")
+	for _, val := range s {
+		if _, err := strconv.Atoi(string(val)); err == nil {
+			fmt.Printf("%q looks like a number.\n", val)
+			userId = val
+			break
+		}
+	}
+
+	return userId, nil
+}
+
+func (r *ThreeScaleAPIClientImpl) VerifyUserIsAdmin(userID string) (bool, error) {
```

Comment:
can't this be done by getting the user and verifying the role assigned to the given user?

Reply:
As discussed, the approach taken here was to follow an existing precedent i.e. the approach taken in 3scale_crudl.go